Framework/DataObjects/inc/MantidDataObjects/ScanningWorkspaceBuilder.h

```diff
@@ -54,50 +54,50 @@ namespace DataObjects {
 */
 class MANTID_DATAOBJECTS_DLL ScanningWorkspaceBuilder {
 public:
-  enum class IndexingType { DEFAULT, TIME_ORIENTED, DETECTOR_ORIENTED };
+  enum class IndexingType { Default, TimeOriented, DetectorOriented };
 
-  ScanningWorkspaceBuilder(const size_t nDetectors, const size_t nTimeIndexes,
-                           const size_t nBins);
+  ScanningWorkspaceBuilder(
+      const boost::shared_ptr<const Geometry::Instrument> &instrument,
+      const size_t nTimeIndexes, const size_t nBins);
 
-  void setHistogram(const HistogramData::Histogram &histogram);
+  void setHistogram(const HistogramData::Histogram histogram);
 
-  void setInstrument(
-      const boost::shared_ptr<const Geometry::Instrument> &instrument);
   void setTimeRanges(const std::vector<
-      std::pair<Kernel::DateAndTime, Kernel::DateAndTime>> &timeRanges);
+      std::pair<Kernel::DateAndTime, Kernel::DateAndTime>> timeRanges);
   void setTimeRanges(const Kernel::DateAndTime &startTime,
                      const std::vector<double> &durations);
-  void setPositions(const std::vector<std::vector<Kernel::V3D>> &positions);
-  void setRotations(const std::vector<std::vector<Kernel::Quat>> &rotations);
-  void setInstrumentAngles(const std::vector<double> &instrumentAngles);
+  void setPositions(const std::vector<std::vector<Kernel::V3D>> positions);
+  void setRotations(const std::vector<std::vector<Kernel::Quat>> rotations);
```

Comment:
If you want to move from these they should not be `const` (which is ok, since you are no passing by value).